hugo/content/docs/learn/workflow/write_grammar.md

```diff
@@ -0,0 +1,58 @@
+---
+title: "3. Write the grammar"
+weight: 400
+url: /docs/learn/workflow/write_grammar
+---
+Your Langium project is now setup and ready to be used. The next step is to define the grammar of your language. The grammar is the most important part of your language definition. It defines the syntax of your language and how the language elements are structured.
+
+The grammar is defined in a `.langium` file. Make sure that you have installed the VS Code extension for Langium. This extension provides syntax highlighting and code completion for `.langium` files. Here's the grammar from the Hello-World example that was generated by the Yeoman generator:
+
+```langium
+grammar HelloWorld
+
+hidden terminal WS: /\s+/;
+terminal ID: /[_a-zA-Z][\w]*/;
+
+entry Model: (persons+=Person | greetings+=Greeting)*;
+
+Person:
+    'person' name=ID;
+
+Greeting:
+    'Hello' person=[Person] '!';
+```
+
+Let's go through this one by one:
+
+```langium
+grammar HelloWorld
+```
+
+Before we tell Langium anything about our grammar contents, we first need to give it a name - in this case it's `HelloWorld`. The `langium-cli` will pick this up to prefix any generated services with this name.
+
+```langium
+hidden terminal WS: /\s+/;
+terminal ID: /[_a-zA-Z][\w]*/;
+```
+
+Here we define our two needed terminals for this grammar: The whitespace `WS` and identifier `ID` terminals. Terminals parse a part of our document by matching it against their regular expression. The `WS` terminal parses any whitespace characters with the regex `/\s+/`. This allows us consume whitespaces in our document. As the terminal is declared as `hidden`, the parser will parse any whitespace and discard the results. That way, we don't have to care about how many whitespaces a user uses in their document. Secondly, we define our `ID` terminal. It parses any string that starts with an underscore or letter and continues with any amount of characters that match the `\w` regex token. It will match `Alice`, `_alice`, or `_al1c3` but not `4lice` or `#alice`. Langium is using the JS regex dialect for terminal definitions.
```

Comment:
```suggestion
Here we define our two needed terminals for this grammar: The whitespace `WS` and identifier `ID` terminals. Terminals parse a part of our document by matching it against their regular expression. The `WS` terminal parses any whitespace characters with the regex `/\s+/`. This allows us to consume whitespaces in our document. As the terminal is declared as `hidden`, the parser will parse any whitespace and discard the results. That way, we don't have to care about how many whitespaces a user uses in their document. Secondly, we define our `ID` terminal. It parses any string that starts with an underscore or letter and continues with any amount of characters that match the `\w` regex token. It will match `Alice`, `_alice`, or `_al1c3` but not `4lice` or `#alice`. Langium is using the JS regex dialect for terminal definitions.
```